torch/signal/windows/windows.py

```diff
@@ -0,0 +1,297 @@
+import torch
+from math import sqrt
+
+from torch import Tensor
+from torch._prims_common import is_float_dtype
+from torch._torch_docs import factory_common_args, parse_kwargs, merge_dicts
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+window_common_args = merge_dicts(
+    parse_kwargs(
+        """
+    M (int): the length of the window.
+        In other words, the number of points of the exponential window.
+    sym (bool, optional): If `False`, returns a periodic window suitable for use in spectral analysis.
+        If `True`, returns a symmetric window suitable for use in filter design. Default: `True`.
+"""
+    ),
+    factory_common_args,
+    {"normalization": "The window is normalized to 1 (maximum value is 1). However, the 1 doesn't appear if "
+                      "`M` is even and `sym` is `True`."}
+)
+
+
+def _add_docstr(*args):
```

Comment:
Why not just reuse the existing `_add_docstr`?

Reply:
The conventional `_add_docstr` is under the `torch._C` namespace, which I reckon doesn't work for Python functions. I tried it before and I get this error (trying to follow the `_torch_docs.py` way of adding docstrings).

Input:

```python
def dummy(a):
    print(a)


_adddocstr(
    dummy,
    r"""
    Ha1
    """
)
```


Output:

```python
TypeError: don't know how to add docstring to type 'function'
```